flask_lastuser/__init__.py

```diff
@@ -400,14 +423,15 @@ def after_request(self, response):
                     httponly=True)                                            # Don't allow reading this from JS.
         return response
 
-    def requires_login(self, f):
+    def requires_login(self, f, scope="*"):
```

Comment:
This won't work. A decorator receives a single parameter, the decorated function. It can't take a second. The API currently looks like this:

```python
def requires_login(f):
    @wraps(f)
    def decorated_function(*args, **kwargs):
        # Insert login check here
        return f(*args, **kwargs)
    return decorated_function

@requires_login
def view_that_requires_login():
    pass
```

To add a `scope` parameter to the decorator, you either need to make a new decorator with an incompatible API, or do some jugaad:

```python
def requires_login(scope='*'):
    def decorator(f):
        @wraps(f)
        def decorated_function(*args, **kwargs):
            # Insert login check here
            return f(*args, **kwargs)
        return decorated_function
    if callable(scope):  # We got a function instead of a scope
        return decorator(scope)
    else:
        return decorator

# Now we can use it with both APIs. Old (backward compatible):
@requires_login
def view_that_requires_login():
    pass

# New (with scope specified):
@requires_login(scope='image')
def view_that_requires_scope():
    pass
```

Reply:
this was an oversight from my side. Just read a little more on decorators. One question - 

> make a new decorator with an incompatible API
But what you did above makes things backward compatible, right? I think the same is done for `requires_permission` below this decorator.